File: backend/src/main/java/com/eventflow/platform/entity/Event.java
package com.eventflow.platform.entity;

import com.eventflow.platform.enums.EventStatus;
import jakarta.persistence.CollectionTable;
import jakarta.persistence.Column;
import jakarta.persistence.ElementCollection;
import jakarta.persistence.Entity;
import jakarta.persistence.EnumType;
import jakarta.persistence.Enumerated;
import jakarta.persistence.FetchType;
import jakarta.persistence.JoinColumn;
import jakarta.persistence.ManyToOne;
import jakarta.persistence.Table;
import java.time.OffsetDateTime;
import java.util.HashSet;
import java.util.Set;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

@Getter
@Setter
@Builder
@NoArgsConstructor
@AllArgsConstructor
@Entity
@Table(name = "events")
public class Event extends BaseEntity {

    @Column(nullable = false, unique = true, length = 220)
    private String slug;

    @Column(nullable = false, length = 200)
    private String title;

    @Column(nullable = false, length = 300)
    private String summary;

    @Column(nullable = false, columnDefinition = "TEXT")
    private String description;

    @Column(name = "cover_image_url", length = 500)
    private String coverImageUrl;

    @Column(name = "location_name", nullable = false, length = 200)
    private String locationName;

    @Column(length = 255)
    private String address;

    @Column(nullable = false, length = 100)
    private String city;

    @Column(name = "start_time", nullable = false)
    private OffsetDateTime startTime;

    @Column(name = "end_time", nullable = false)
    private OffsetDateTime endTime;

    @Column(name = "registration_deadline", nullable = false)
    private OffsetDateTime registrationDeadline;

    @Column(nullable = false)
    private Integer capacity;

    @Column(name = "confirmed_count", nullable = false)
    private Integer confirmedCount;

    @Column(name = "waitlist_count", nullable = false)
    private Integer waitlistCount;

    @Column(nullable = false)
    private Boolean featured;

    @Enumerated(EnumType.STRING)
    @Column(nullable = false, length = 20)
    private EventStatus status;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "category_id")
    private Category category;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "created_by")
    private User createdBy;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "updated_by")
    private User updatedBy;

    @Column(name = "published_at")
    private OffsetDateTime publishedAt;

    @Column(name = "cancelled_at")
    private OffsetDateTime cancelledAt;

    @Builder.Default
    @ElementCollection(fetch = FetchType.EAGER)
    @CollectionTable(name = "event_tags", joinColumns = @JoinColumn(name = "event_id"))
    @Column(name = "tag", nullable = false, length = 50)
    private Set<String> tags = new HashSet<>();
}

File: frontend/src/types/event.ts
export type EventStatus = 'DRAFT' | 'PUBLISHED' | 'CLOSED' | 'CANCELLED';

export interface EventSummary {
  id: number;
  slug: string;
  title: string;
  summary: string;
  coverImageUrl?: string;
  city: string;
  locationName: string;
  startTime: string;
  endTime: string;
  registrationDeadline: string;
  capacity: number;
  confirmedCount: number;
  remainingSeats: number;
  featured: boolean;
  status: EventStatus;
  categoryCode: string;
  tags: string[];
}

export interface CurrentUserRegistration {
  state: 'CONFIRMED' | 'WAITING' | 'NONE' | 'CANCELLED';
  bookingId?: number | null;
  waitlistEntryId?: number | null;
  waitlistPosition?: number | null;
}

export interface EventDetail extends EventSummary {
  description: string;
  address?: string;
  waitlistCount: number;
  bookable: boolean;
  bookableReason?: string | null;
  currentUserRegistration?: CurrentUserRegistration | null;
}

export interface EventQuery {
  keyword?: string;
  category?: string;
  city?: string;
  startDateFrom?: string;
  startDateTo?: string;
  sort?: string;
  page?: number;
  size?: number;
}

export interface EventUpsertRequest {
  title: string;
  summary: string;
  description: string;
  coverImageUrl?: string;
  locationName: string;
  address?: string;
  city: string;
  startTime: string;
  endTime: string;
  registrationDeadline: string;
  capacity: number;
  featured: boolean;
  categoryCode: string;
  tags: string[];
}
